nibabel/streamlines/array_sequence.py

```diff
@@ -0,0 +1,256 @@
+import numpy as np
+
+
+class ArraySequence(object):
+    """ Sequence of ndarrays having variable first dimension sizes.
+
+    This is a container allowing to store multiple ndarrays where each ndarray
+    might have different first dimension size but a *common* size for the
+    remaining dimensions.
+
+    More generally, an instance of :class:`ArraySequence` of length $N$ is
+    compoosed of $N$ ndarrays of shape $(d_n, d_2, ... d_D)$ where
+    $n \in [1,N]$, $d_n$ might vary from one ndarray to another and
+    $d_2, ..., d_D)$ have to be the same for every ndarray.
+
+    """
+
+    BUFFER_SIZE = 87382*4  # About 4 Mb if item shape is 3 (e.g. 3D points).
+
+    def __init__(self, iterable=None):
+        """
+        Parameters
+        ----------
+        iterable : None or iterable or :class:`ArraySequence`, optional
+            If None, create an empty :class:`ArraySequence` object.
+            If iterable, create a :class:`ArraySequence` object initialized
+            from array-like objects yielded by the iterable.
+            If :class:`ArraySequence`, create a view (no memory is allocated).
+            For an actual copy use :meth:`.copy` instead.
+
+        """
+        # Create new empty `ArraySequence` object.
+        self._data = np.array(0)
+        self._offsets = np.array([], dtype=np.intp)
+        self._lengths = np.array([], dtype=np.intp)
+
+        if isinstance(iterable, ArraySequence):
+            # Create a view.
+            self._data = iterable._data
+            self._offsets = iterable._offsets
+            self._lengths = iterable._lengths
```

Comment:
To avoid a lot of indenting - how about:

```
          self._lengths = iterable._lengths
          return
    if iterable is None:
        return # could also go above the first isinstance check.
```